WPMedia/fusion-news-theme-blocks
# TMEDIA-747 : Small Manual Promo Block

## Description

1. `small-manual-promo-block` has been migrated to the Themes 2.0 architecture.
2. Testing was updated to use React Testing Library.
3. Readme was updated.

## Jira Ticket

- [TMEDIA-747 : Small Manual Promo Block](https://arcpublishing.atlassian.net/browse/TMEDIA-747)

## Acceptance Criteria

No special notes in Jira ticket. Followed [checklist](https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3106243821/Converting+a+v1+block+to+v2+architecture) for migrating blocks from v1 to v2.

## Test Steps

### Blocks repository and testing with Storybook.
1. Checkout this branch `git checkout TMEDIA-747-small-manual-promo-block`
2. Run storybook with `npm run storybook` and verify functionality corresponding with stories.
    a. Note: Stories displaying different image aspect ratios should be disregarded, pending future work with Image Resizer.

### Block Appearance
(as seen with Firefox grid view)

#### Image Position: Left
<img width="886" alt="image" src="https://user-images.githubusercontent.com/452134/175136488-bd4df880-91a4-4887-b42b-807e4730a484.png">

#### Image Position: Right
<img width="888" alt="image" src="https://user-images.githubusercontent.com/452134/175136809-05a8175c-e9b1-4765-b27f-2d95bef1a729.png">

#### Image Position: Above
<img width="886" alt="image" src="https://user-images.githubusercontent.com/452134/175137097-0d9fa2b5-8e46-4068-aacd-eaa5603c8185.png">

#### Image Position: Below
<img width="885" alt="image" src="https://user-images.githubusercontent.com/452134/175137794-6812449e-3b20-46e2-a885-46a7d346ac97.png">

#### Fallback Image
The `arcSite` fallback image is used if `showImage` is selected but the `imageURL` custom field is a falsy value.
<img width="881" alt="image" src="https://user-images.githubusercontent.com/452134/175697268-e1ad7fc7-3d19-4e45-8954-4000fee29f77.png">
 
### Testing with Feature Pack repository
1. Change the branch of your local `arc-themes-feature-pack` repository to `TMEDIA-747-small-manual-promo-block`.
2. Set `THEMES_BLOCKS_REPO` in the `.env` file to point to your local `arc-themes-blocks` repository.
4. Start fusion while linking to local block with `npx fusion start -f -l @wpmedia/small-manual-promo-block`
5. Verify cosmetic and functional parity.

## Other Notes
1. When horizontal layout is used on the block, CSS Grid was used to maintain the space occupied the the Heading and MediaItem. The Heading area should always have twice the horizontal space of the Image.
2. The [Figma card for small blocks](https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?node-id=3507%3A32495) was updated to show the gap difference in horizontal, vertical presentation.
3. Respective PR in `arc-themes-feature-pack` can be found at https://github.com/WPMedia/arc-themes-feature-pack/pull/281 .

## Dependencies or Side Effects

1. `small-manual-promo-block` still uses `@wpmedia/engine-theme-sdk` for `LazyLoad`.

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
